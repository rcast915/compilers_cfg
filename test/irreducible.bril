@main() {
  v0: int = const 0;
  v1: int = const 1;

.entry:
  v2: bool = lt v0 v1;
  br v2 .A .B;

.A:
  jmp .C;

.B:
  jmp .D;

.C:
  v3: bool = lt v0 v1;
  br v3 .B .exit;     # C -> B (into the cycle)

.D:
  v4: bool = lt v0 v1;
  br v4 .A .exit;     # D -> A (into the cycle)

.exit:
  ret;
}
