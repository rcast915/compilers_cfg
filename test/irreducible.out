digraph main {
 b0;
 entry;
 A;
 B;
 C;
 D;
 exit;
 b0 -> entry;
 entry -> A;
 entry -> B;
 A -> C;
 B -> D;
 C -> B;
 C -> exit;
 D -> A;
 D -> exit;
}
paths: {'b0': 0, 'entry': 1, 'A': 2, 'B': 2, 'C': 3, 'D': 3, 'exit': 4}
rpo: ['b0', 'entry', 'A', 'C', 'B', 'D', 'exit']
backs: [('D', 'A')]
reducible: False
